### PR TITLE
imp(ante): Refactor ante handlers to be easier to use in partner chains (target main)

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR is superseding #51 after cherry-picking the commits on top of the `main` branch.

I've moved the `MonoEVMDecorator` into the general utils and out of the chain definition, because it includes all required checks for EVM transactions that also our partners will want to run.

On top of that, there were some outdated required keepers (staking & distribution) which were required when we had to logic to auto-withdraw rewards for fee payments.

Other than those things, there are some minor refactors / improvements.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `MonoDecorator` for enhanced Ethereum transaction validation and processing.
	- Added a post-handler for transaction processing in the ExampleChain application.
	- Expanded project structure with new packages and directories for improved functionality.

- **Bug Fixes**
	- Enhanced error handling in the Keeper's chain ID management for clearer panic messages.

- **Refactor**
	- Streamlined ante handler initialization and gas consumption processes.
	- Updated account management interfaces to simplify dependencies.
	- Removed unnecessary keeper dependencies from various components.

- **Tests**
	- Improved test cases for gas consumption and ante handler options validation.

- **Chores**
	- Updated changelog to reflect all recent changes and improvements.

### CODE REVIEW

File: ante/evm/08_gas_consume.go [hunk 1]
```diff
@@ -72,21 +65,22 @@ func ConsumeFeesAndEmitEvent(
 // If the spendable balance is not enough, it tries to claim enough staking rewards to cover the fees.
```
Comment (MalteHerrmann) [inline ante/evm/08_gas_consume.go:65]: adjust comment here, it's outdated

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ This changelog was created using the `clu` binary
 
 ### Improvements
 
+- (ante) [#51](https://github.com/evmos/os/pull/51) Refactor ante handlers to be easier to use in partner chains.
```
Comment (MalteHerrmann) [inline CHANGELOG.md:11]:
```suggestion
- (ante) [#52](https://github.com/evmos/os/pull/52) Refactor ante handlers to be easier to use in partner chains.
```

File: ante/evm/mono_decorator.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+		txFee := UpdateCumulativeTxFee(
+			decUtils.TxFee,
+			txData.Fee(),
+			decUtils.EvmDenom,
+		)
+		decUtils.TxFee = txFee
+		decUtils.TxGasLimit += gas
+
+		// 10. increment sequence
```
Comment (MalteHerrmann) [inline ante/evm/mono_decorator.go:205]: update numbers here

**Review by 0xstepit — APPROVED** [review]
Great job @MalteHerrmann 💪

**Review by hanchon — APPROVED** [review]

**Pull request merged**